apps/settings/js/panels/addons/addons.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Handle addons panel's functionality.
+ *
+ * @module AddonsManager
+ */
+define(function(require) {
+  'use strict';
+
+  var AppsCache = require('modules/apps_cache');
+  var Observable = require('modules/mvvm/observable');
+
+  var AddonsManager = function am() {
+    this.addons = Observable({
+      apps: []
+    });
+  };
+
+  AddonsManager.prototype = {
+    /**
+     * initialization
+     *
+     * @memberOf AddonsManager
+     * @access public
+     */
+    init: function am_init() {
+      var self = this;
+      return AppsCache.apps().then(function(apps) {
+        apps.some(function(app) {
+          if (self._isAddon(app)) {
+            app.addonManager = self;
+            self.addons.apps.push(app);
+          }
+        });
+      });
+    },
+
+    set enabled(value) {
+      if (value !== this._enabled) {
+        this._enabled = value;
+        if (this._enabled) {
+          this._bindEvents();
+        } else {
+          this._unbindEvents();
+        }
+      }
+    },
+
+    _bindEvents: function am__bindEvents() {
+      AppsCache.addEventListener('oninstall', this._updateAddons.bind(this));
+      AppsCache.addEventListener('onuninstall', this._updateAddons.bind(this));
+    },
+
+    _unbindEvents: function am__unbindEvents() {
+      AppsCache.removeEventListener('oninstall', this._updateAddons);
+      AppsCache.removeEventListener('onuninstall', this._updateAddons);
+    },
+
+    /**
+     * We have to update the addon count based on incoming evt and
+     * decide to show/hide or not.
+     *
+     * @param {Object} evt
+     * @memberOf AddonsManager
+     */
+    _updateAddons: function(evt) {
+      var app = evt && evt.application;
+      var type = evt.type;
+
+      if (this._isAddon(app)) {
+        if (type === 'install') {
+          app.addonManager = this;
+          this.addons.apps.push(app);
+        } else if (type === 'uninstall') {
+          var index = this.addons.apps.indexOf(app);
+          if (index !== -1) {
+            this.addons.apps.splice(index, 1);  
+          }
+        }
+      }
+    },
+
+    /**
+     * Check whether this app is an addon
+     *
+     * @param {Object} app
+     * @returns {Boolean}
+     * @memberOf AddonsManager
+     */
+    _isAddon: function(app) {
+      var manifest = app.manifest || app.updateManifest;
+      return manifest.role === 'addon';
+    },
+
+    enableAddon: function(app) {
+      navigator.mozApps.mgmt.setEnabled(app, true);
+    },
+
+    disableAddon: function(app) {
+      navigator.mozApps.mgmt.setEnabled(app, false);
+    }
+
+  };
+
+  return function ctor_addons_manager() {
+    return new AddonsManager();
```

Comment:
We could make `AddonsManager` an `Observable` then we are able to notify the UI part (`addons_item.js`) with the change of the addon length. Implementation details please refer to this[1].
[1]: https://github.com/mozilla-b2g/gaia/blob/master/apps/settings/js/panels/display/wallpaper.js#L142